src/CrateTokenV1.sol

```diff
@@ -54,100 +44,123 @@ contract CrateTokenV1 is ERC20Upgradeable, ReentrancyGuard {
         artistFeeDestination = _artistAddress;
         protocolFeeDestination = _protocolAddress;
         uniswapV2Router02 = _uniswapV2Router02;
-        bondingCurveActive = true;
-
+        phase = Phase.CROWDFUND;
         tokensInCurve = MAX_CURVE_SUPPLY;
         songURI = _songURI;
         _approve(address(this), uniswapV2Router02, MAX_SUPPLY);
     }
 
-    function buyWithEth(uint256 minTokensReceivable) external payable {
-        // Take fees out of ETH, then see how many tokens you can buy with the remaining amount.
-        require(bondingCurveActive, "Bonding curve ended");
-        require(tokensInCurve > 0, "Bonding curve is sold out");
+    /// PUBLIC ///
 
+    function buyWithEth(uint256 minTokensReceivable) external payable {
         uint256 preFee =
             (msg.value * (CRATE_FEE_PERCENT + ARTIST_FEE_PERCENT)) / (CRATE_FEE_PERCENT + ARTIST_FEE_PERCENT + 1 ether);
-        uint256 netValue = msg.value - preFee;
+        uint256 netValue = ((msg.value - preFee) * 999) / 1000;
```

Comment:
what is this part? `* 999) / 1000`

Reply:
Tests kept failing due to precision loss - at a higher level I'd suggest removing the `buyWithEth` function entirely but I can see the appeal